packages/api/src/utils/stripe.js

```diff
@@ -256,7 +258,26 @@ export class StripeCustomersService {
       name: options?.name,
       email: options?.email
     })
-    await this.userCustomerService.upsertUserCustomer(user.id.toString(), createdCustomer.id)
+
+    try {
+      await this.userCustomerService.insertUserCustomer(user.id.toString(), createdCustomer.id)
+    } catch (error) {
+      // due to potential race conditions, we want to ensure we only return an actual associated stripe customer
+      // and remove stripe customers that we fail to associate
+      if (hasOwnProperty(error, 'code') && error.code === 'CONSTRAINT_ERROR_DB') {
```

Comment:
Try to hide the implementation details inside `insertUserCustomer` interface.

e.g. detecting a `UserAlreadyHasCustomer` error. `insertUserCustomer`, when implementation is powered by a db, can throw it when its underlying db sees this error

Reply:
doing try/catch with a more abstract error is a good idea regardless to hide the db details from this stripe implementation service that otherwise is abstracted away via the 'service' interfaces that can be injected in.

I would *also* encourage then avoiding try/catch alltogether so you can also have the type checker make sure callers handle that case where insertUserCustomer cannot complete due to the uniqueness constraint that is inherent in our domain model. So personally I'd also recommend having `insertUserCustomer` return `Promise<void | UserAlreadyHasCustomer>`